deploy/helm/elastic-agent/templates/integrations/_auto_ops/secret.yaml

```
@@ -0,0 +1,33 @@
+{{- if eq $.Values.autoOps.enabled true }}
+{{- $presetName := "autoOps" }}
+{{- $presetVal := $.Values.agent.presets.autoOps }}
+{{- $agentName := include "elasticagent.preset.fullname" (list $ $presetName) }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ $agentName }}-autoops
+  namespace: {{ .namespace | default $.Release.Namespace | quote }}
+  labels:
+    {{- include "elasticagent.labels" $ | nindent 4 }}
+    {{- with ($presetVal).labels -}}
+    {{ toYaml . | nindent 4 }}
+    {{- end }}
+  {{- with ($presetVal).annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+stringData:
+  autoops-token: {{ $.Values.autoOps.autoops_token | quote }}
+  {{- if and $.Values.autoOps.es_username $.Values.autoOps.es_password }}
+  es-username: {{ $.Values.autoOps.es_username | quote }}
+  es-password: {{ $.Values.autoOps.es_password | quote }}
+  {{- end }}
+  autoops-es-url: {{ $.Values.autoOps.autoops_es_url | quote }}
+  temp-resource-id: {{ $.Values.autoOps.autoops_temp_resource_id | quote }}
+  otel-url: {{ $.Values.autoOps.autoops_otel_url | quote }}
+  {{- if $.Values.autoOps.es_api_key }}
+  es-api-key: {{ $.Values.autoOps.es_api_key | quote }}
+  {{- end }}
+  cloud-connected-mode-api-key: {{ $.Values.autoOps.elastic_cloud_connected_mode_api_key | quote }}
+  cloud-connected-mode-api-url: {{ $.Values.autoOps.elastic_cloud_connected_mode_api_url | quote }}
```

since you allow empty strings for `autoops_es_url`, `elastic_cloud_connected_mode_api_key` and `elastic_cloud_connected_mode_api_url` these needed to be wrapped in if's

```suggestion
  autoops-token: {{ $.Values.autoOps.autoops_token | quote }}
  {{- if and $.Values.autoOps.es_username $.Values.autoOps.es_password }}
  es-username: {{ $.Values.autoOps.es_username | quote }}
  es-password: {{ $.Values.autoOps.es_password | quote }}
  {{- end }}
  {{- if $.Values.autoOps.autoops_es_url }}
  autoops-es-url: {{ $.Values.autoOps.autoops_es_url | quote }}
  {{- end }}
  temp-resource-id: {{ $.Values.autoOps.autoops_temp_resource_id | quote }}
  otel-url: {{ $.Values.autoOps.autoops_otel_url | quote }}
  {{- if $.Values.autoOps.es_api_key }}
  es-api-key: {{ $.Values.autoOps.es_api_key | quote }}
  {{- end }}
  {{- if $.Values.autoOps.elastic_cloud_connected_mode_api_key }}
  cloud-connected-mode-api-key: {{ $.Values.autoOps.elastic_cloud_connected_mode_api_key | quote }}
  {{- end }}
  {{- if $.Values.autoOps.elastic_cloud_connected_mode_api_url }}
  cloud-connected-mode-api-url: {{ $.Values.autoOps.elastic_cloud_connected_mode_api_url | quote }}
  {{- end }}
```